editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-strategy-parent-lookup.ts

```diff
@@ -0,0 +1,482 @@
+import { SimpleFlexDirection } from '../../../../../core/layout/layout-utils'
+import { MetadataUtils } from '../../../../../core/model/element-metadata-utils'
+import { getStoryboardElementPath } from '../../../../../core/model/scene-utils'
+import { mapDropNulls } from '../../../../../core/shared/array-utils'
+import * as EP from '../../../../../core/shared/element-path'
+import {
+  ElementInstanceMetadata,
+  ElementInstanceMetadataMap,
+} from '../../../../../core/shared/element-template'
+import {
+  CanvasPoint,
+  CanvasRectangle,
+  rectContainsPoint,
+  rectContainsPointInclusive,
+  Size,
+  size,
+  sizeFitsInTarget,
+} from '../../../../../core/shared/math-utils'
+import { ElementPath } from '../../../../../core/shared/project-file-types'
+import { AllElementProps } from '../../../../editor/store/editor-state'
+import { getAllTargetsAtPointAABB } from '../../../dom-lookup'
+import { InteractionCanvasState } from '../../canvas-strategy-types'
+import { AllowSmallerParent } from '../../interaction-state'
+import {
+  SingleAxisAutolayoutContainerDirections,
+  singleAxisAutoLayoutContainerDirections,
+} from '../flow-reorder-helpers'
+import { ReparentStrategy, ReparentSubjects, ReparentTarget } from './reparent-strategy-helpers'
+import { drawTargetRectanglesForChildrenOfElement } from './reparent-strategy-sibling-position-helpers'
+
+export type FindReparentStrategyResult = {
+  strategy: ReparentStrategy
+  isFallback: boolean
+  target: ReparentTarget
+}
+
+export function getReparentTargetUnified(
```

Comment:
this is the same behavior as the old massive spaghetti-function getReparentTargetUnified, but with the code split into sub-functions for better readability